src/executor/admin/SubmitJobExecutor.cpp

```diff
@@ -114,22 +85,46 @@ folly::Future<Status> SubmitJobExecutor::execute() {
                     return finish(std::move(v));
                 }
                 case meta::cpp2::AdminJobOp::SHOW_All: {
-                    nebula::DataSet v({"Job Id", "Command", "Status", "Start Time", "Stop Time"});
                     DCHECK(resp.value().__isset.job_desc);
                     if (!resp.value().__isset.job_desc) {
                         return Status::Error("Response unexpected");
                     }
                     const auto &jobsDesc = *resp.value().get_job_desc();
-                    for (const auto &jobDesc : jobsDesc) {
-                        v.emplace_back(nebula::Row({
-                            jobDesc.get_id(),
-                            meta::cpp2::_AdminCmd_VALUES_TO_NAMES.at(jobDesc.get_cmd()),
-                            meta::cpp2::_JobStatus_VALUES_TO_NAMES.at(jobDesc.get_status()),
-                            jobDesc.get_start_time(),
-                            jobDesc.get_stop_time(),
-                        }));
+                    auto spaceName = qctx()->rctx()->session()->space().name;
+                    if (cmd == meta::cpp2::AdminCmd::REBUILD_TAG_INDEX ||
+                        cmd == meta::cpp2::AdminCmd::REBUILD_EDGE_INDEX) {
+                        std::unordered_map<std::string, meta::cpp2::JobStatus> indexStatus;
```

Comment:
@Shylock-Hg cmd is needed here

Reply:
You still capture it, move these code to the callback make this callback contains too many code.